### PR TITLE
feature: Extract true sensor width and focal length from EXIF

### PR DESCRIPTION
**Pull request by GITNE**
Most dedicated cameras report the focal length and sensor size in the EXIF header. Traditionally, only few smartphones have been reporting this data but with the latest smartphones on the market, more and more devices report this crucial data too. This data should finally enable correct camera calibration and ultimately also improve reconstruction quality (mainly precision with increasing distance).

For this feature to work the EXIF header must have the `FocalLength`, `FocalPlaneXResolution`, and `FocalPlaneResolutionUnit` tags. Meta data extraction still falls back to database values on known cameras which do not report the required EXIF tags or to the default focal ratio (`default_focal_prior`) if the camera could not be found in the database. Please note that many of the sensor size values in the database are skewed or copied from marketing material instead of technical documentation. Anyhow, the values in the EXIF header should always be preferred over any database values.

### CODE REVIEW

**Comment by GITNE** [comment]
Oh, I forgot to mention that finally *`unknown`* cameras, like new ones or which simply are not in the database, get the correct focal ratio as long as there is proper EXIF data in the image file.

**Comment by GITNE** [comment]
@paulinus Ping? I am aware that you folks are quite busy but I think that this PR is also quite straight forward and does not break anything. If you seem to be too busy, my tip would be to try to find some time for PR review at least once a week, half a day or so. Well, this used to work for me :wink:

File: opensfm/exif.py [hunk 1]
```diff
@@ -176,10 +181,25 @@ def extract_focal(self):
         focal_35, focal_ratio = compute_focal(
             get_tag_as_float(self.tags, 'EXIF FocalLengthIn35mmFilm'),
             get_tag_as_float(self.tags, 'EXIF FocalLength'),
-            get_tag_as_float(self.tags, 'EXIF CCD width'),
+            self.extract_sensor_width(),
```
Comment (paulinus) [inline opensfm/exif.py]: By doing this, we are no longer reading the `EXIF CCD width` tag.  Should `extract_sensor_width()` handle the `EXIF CCD width` tag?

Comment (paulinus) [reply opensfm/exif.py]: Awesome, wonder how we got look for it. Was just checking now and could not find any reference of it anywhere. Thanks for looking at this.

**Comment by paulinus** [comment]
@GITNE, i've refactored the function to accept also other units https://github.com/mapillary/OpenSfM/commit/a074feab193776569ff23ce0ff5cab31535e34ff
I've followed the unit definitions in https://www.sno.phy.queensu.ca/~phil/exiftool/TagNames/EXIF.html
I do not have image examples using other units to test it on. Please, let me know if that looks correct to you. Thanks.

**Comment by GITNE** [comment]
+1 Alright, looks good to me.

**Pull request merged**